ms-common-impl/src/main/java/net/trajano/ms/vertx/jaxrs/JwtAssertionInterceptor.java

```diff
@@ -76,81 +68,51 @@
-                final JWEObject jwe = (JWEObject) joseObject;
-                jwe.decrypt(new RSADecrypter(jwksProvider.getDecryptionKey(
-                    jwe.getHeader().getKeyID())));
-                joseObject = JOSEObject.parse(jwe.getPayload().toString());
-            }
-
-            if (joseObject instanceof JWSObject) {
-                final JWSObject jws = (JWSObject) joseObject;
-
-                final URI signatureJwksUri = jwksUriProvider.getUri(requestContext);
-                if (signatureJwksUri != null) {
-                    final JWSVerifier verifier = new RSASSAVerifier(getSigningKey(jws.getHeader().getKeyID(), signatureJwksUri));
-                    if (!jws.verify(verifier)) {
-                        LOG.warn("JWT verification failed for {}", requestContext.getUriInfo());
-                        requestContext.abortWith(Response.status(Status.UNAUTHORIZED)
-                            .header(HttpHeaders.WWW_AUTHENTICATE, "JWT")
-                            .entity("signature vertification failed")
-                            .build());
-                        return;
-                    }
+            HttpsJwks httpsJwks;
+            final String jwksUri = requestContext.getHeaderString(X_JWKS_URI);
+            if (jwksUri == null) {
+                httpsJwks = null;
+            } else {
+                httpsJwks = jwks.get(jwksUri);
+                if (httpsJwks == null) {
+                    httpsJwks = new HttpsJwks(jwksUri);
                 }
             }
-            claims = JWTClaimsSet.parse(joseObject.getPayload().toString());
-        } catch (final ParseException
-            | JOSEException
-            | ExecutionException e) {
-            throw new BadRequestException("unable to parse JWT");
-        }
-        if (audience != null && !claims.getAudience().contains(audience.toASCIIString())) {
-            LOG.warn("Audience {} did not match {} for {}", claims.getAudience(), audience, requestContext.getUriInfo());
-            requestContext.abortWith(OAuthTokenResponse.unauthorized("invalid_audience", "Audience validation failed", "Bearer").getResponse());
-            return;
-
-        }
-        if (issuer != null && !claims.getIssuer().equals(issuer.toASCIIString())) {
-            LOG.warn("Issuer {} did not match {} for {}", claims.getIssuer(), issuer, requestContext.getUriInfo());
-            requestContext.abortWith(OAuthTokenResponse.unauthorized("invalid_issuer", "Issuer validation failed", "Bearer").getResponse());
-            return;
-        }
-
-        if (claims.getExpirationTime() != null && claims.getExpirationTime().before(requestContext.getDate())) {
-            LOG.warn("Claims expired for {}", requestContext.getUriInfo());
-            requestContext.abortWith(OAuthTokenResponse.unauthorized("invalid_claims", "Claims expired", "Bearer").getResponse());
+            final String audience = requestContext.getHeaderString(X_JWT_AUDIENCE);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "audience" which hides the field declared at line 59. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)